File: Data Structures & Algorithms/car-fleet/submission-2.js
class Solution {
    /**
     * @param {number} target
     * @param {number[]} position
     * @param {number[]} speed
     * @return {number}
     */
    carFleet(target, position, speed) {
        let arr = position.map((key, index) => [key, speed[index]])
        arr = arr.sort((a, b) => a[0]-b[0])
        arr = arr.map(key => ((target - key[0])/key[1]))
        console.log(arr)

        let curr = -Infinity
        let res = 0
        for(let key of position){
            let val = arr.pop()
            if ( val <= curr){
                console.log(val)
            }
            else{
                curr = val
                res ++
            }
        }
        return res
    }
}

File: Data Structures & Algorithms/non-overlapping-intervals/submission-2.py
class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        intervals.sort()
        ans = 0
        curr = intervals[0]
        i = 1
        while i < len(intervals):
            if curr[1] > intervals[i][1]:
                curr = intervals[i]
                ans += 1
            elif curr[1] <= intervals[i][0]:
                curr = intervals[i]
            else:
                ans += 1
            i += 1

        return ans

